test/components/util/util.go

```diff
@@ -183,6 +183,11 @@ func WaitForDeployment(t *testing.T, client kubernetes.Interface, ns, name strin
 			return false, err
 		}
 		pods = filterNonControllerPods(pods)
+
+		if len(pods.Items) == 0 {
+			t.Fatalf("checking containers status failed. No pods selected.")
```

Comment:
> WaitForDeployment now gives deployment.Metadata.Labels for selecting the
> pods of the deployment, but they may not always be the same as labels

What do you mean by gives? And this deviant behaviour is consistent or intermittent and this behaviour depends on what exactly?

Reply:
>What do you mean by gives? 

`s/gives/uses`.

>And this deviant behaviour is consistent or intermittent and this behaviour depends on what exactly?

This depends on the specification of tested deployment. Deployment labels != deployment pod labels.

Reply:
I rephrased commit messages.